src/sources.js

```diff
@@ -294,13 +294,18 @@ class S3Source extends Source {
     return `https://${this.bucket}.s3.amazonaws.com`;
   }
   async _listObjects() {
-    // XXX TODO: This will only return the first 1000 objects.  That's fine for
-    // now (for comparison, nextstrain-data only has ~500), but we really
-    // should iterate over the whole bucket contents using the S3 client's
-    // pagination support.
-    //   -trs, 30 Aug 2019
-    const list = await S3.listObjectsV2({Bucket: this.bucket}).promise();
-    return list.Contents;
+    const s3Params = {Bucket: this.bucket, MaxKeys: 1000};
+    let contents = [];
+    let existMoreKeys = true;
+    let lastKeySeen = false;
+    while (existMoreKeys) {
+      // eslint-disable-next-line no-await-in-loop
+      const list = await S3.listObjectsV2(lastKeySeen ? {...s3Params, StartAfter: lastKeySeen} : s3Params).promise();
+      existMoreKeys = list.IsTruncated;
+      contents = [...contents, ...list.Contents];
```

Comment:
we can avoid iterating over both `contents` and `list.Contents` here by instead writing
```
contents = contents.concat(list.Contents)
```

Reply:
Done. Thanks!